### ISSUE TITLE
Const variable to get version of VP editor the table was created in

### ISSUE THREAD

**Issue by Rascal5688** [vpinball/vpinball#293]
**Is your feature request related to a problem? Please describe.**
Other then the warning about the table being made in newer version of VP, it would be nice to have a const variable that could be read via script where an author or an external program can get the full version of the editor the table was last saved in (10.8.0.782, etc...). If it could be stamped in the vpx file somewhere and readable from an external launcher or sorting program or script, so it could be used to launch the table in the version of the editor/player it was intended to run on. I know that it is possible to grab the 10.6, 10.7, and 10.8, but the full version would be more helpful. Thank you.

**Comment by toxieainc** [comment]
I don't think we really want that, as all versions that are not finals are considered beta anyway, and thus are potentially also unstable/not-ready-for-prime-time (e.g. as new stuff that is saved with the table may still be changing again).
Or do you just mean the subversions (e.g. 10.7.X like 10.7.1, 10.7.2, 10.7.3)?

**Comment by Rascal5688** [comment]
> I don't think we really want that, as all versions that are not finals are considered beta anyway, and thus are potentially also unstable/not-ready-for-prime-time (e.g. as new stuff that is saved with the table may still be changing again). Or do you just mean the subversions (e.g. 10.7.X like 10.7.1, 10.7.2, 10.7.3)?

What you are saying makes sense. I guess I should not be releasing any tables that are not made with official releases of VPX (GL or not). Subversion I think we can get already for the most part, it could be simpler maybe via a const variable. What we get sometimes is a table made in one version, opened and saved in another which retags it to the last saved version. Which if someone is rolling back may cause the table to error on like a newly added feature like "RenderingMode" for example. Authors like me like this feature in order for the table to know when it should be in VR mode via script. The table crashes with an error when it doesn't include "RenderingMode" in the editor. Also, it's helpful for a frontend loader to know what version of VPX to launch a certain table. I'm not sure how we can beta test a version of VPX without making a table and using these new awesome features. Full version tagging, beta or not, could be used or not used I suppose, by whoever needs or doesn't need it, but if it was there it could be used. If its a lot of work, forget it, there are far more important things you guys are doing. You guys are the best. You keep this hobby rocking man.

**Comment by Gigalula** [comment]
First sorry for my bad English.

As I can see as I read, I deeply understand your concern that Git releases are not final releases and should not be released and I do assure you It's not my intenssion to act as such. But on the other hand we are simply here to test and evaluate new features and/or detect new possible bugs and a new labeling format with more information could lead to better help in the future when it's time to report.

I will try to explain as much as I can in English:

Table use as example:

Dirty Harry (Williams 1995) v2.20.vpx

When we get to read the version it says 1072

So far so good, we are using the original release version when it is tested in VR, at start something is wrong with the lights, so I guess this is on of these explananion

Scenario 1: The creator has a good version of 10.7.2 _GL .exe not available so impossible to test in real or good condition

Scenario 2: I think he is using the normal standard VPX and not the GL version to save the table before releasing it.

So if we can have a more detailed TAG including (Version, Build, "VPX or VPX_GL") used when saving the table. This could maybe lead to less false report on git.

Sorry for my bad English but I think it could be more productive and benefic to all. Of course, there is no question of making it public, but serve the cause. I love your work and you are without any doubt a great team and I hope you can consider it positively.

Cordially
